### PR TITLE
add set to the batch-read rec

### PR DESCRIPTION
**Pull request by sol-sh**
add set to the record returned by batch-read

### CODE REVIEW

File: src/main/clojure/aerospike_clj/client.clj [hunk 1]
```diff
@@ -146,9 +147,10 @@
   (Bin/asNull bin-name))
 
 (defn- batch-read->map [^BatchRead batch-read]
-  (assoc (record/record->map (.record batch-read))
-         :index
-         (.toString (.userKey (.key batch-read)))))
+  (let [^Key key (.key batch-read)]
+    (-> (record/record->map (.record batch-read))
+        (assoc :index (.toString (.userKey key)))
+        (assoc :set (.setName key)))))
```
Comment (barkanido) [inline src/main/clojure/aerospike_clj/client.clj:153]: a single `assoc` can take multiple key-value args

File: project.clj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-(defproject aerospike-clj "1.0.0"
+(defproject aerospike-clj "1.0."
```
Comment (barkanido) [inline project.clj]: 1.0.1-SANPSHOT

**Review by barkanido — CHANGES_REQUESTED** [review]
Thanks!
Can you please:

- update the relevant docs
- give a full description for this MR?

File: src/main/clojure/aerospike_clj/client.clj (hunk 1 above)
Comment (sol-sh) [reply src/main/clojure/aerospike_clj/client.clj:153]: yes, but its less performant as it do it by iteration

Comment (sol-sh) [reply src/main/clojure/aerospike_clj/client.clj:153]: and the doc already reflecy my change:

get-batch
(get-batch db batch-reads)
(get-batch db batch-reads conf)
Get a batch of records from the cluster asynchronously. batch-reads is a collection of maps of the form {:index "foo" **:set "someset"** :bins [...]} the :bins key can have required bins for the batched keys or missing/[:all] to get all the bins (see _get). The result is a vector of AerospikeRecords in the same order of keys. Missing keys result in nil in corresponding positions.

File: project.clj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-(defproject aerospike-clj "1.0.0"
+(defproject aerospike-clj "1.0.1-SANPSHOT\n\n"
```
Comment (barkanido) [inline project.clj]: why the double `\n`?

**Review by barkanido — CHANGES_REQUESTED** [review]
can you please update `deftest get-batch` to reflect your changes?

File: project.clj (hunk 3 above)
Comment (sol-sh) [reply project.clj]: no idea, where it came from, fixed

File: src/main/clojure/aerospike_clj/client.clj (hunk 1 above)
Comment (yaronel) [reply src/main/clojure/aerospike_clj/client.clj:153]: @sol-sh, did you check the performance implications of a single vs. multiple `assoc`?
I have, and it's quite unremarkable:

```
 (c/with-progress-reporting
    (c/bench
      (-> {:foo "bar" :baz [10 20] :fizz {:fuzz "bar"}}
          (assoc :index (.toString 1))
          (assoc :set "set-name"))
      :verbose))


  "Evaluation count : 47295360 in 60 samples of 788256 calls.
   Execution time sample mean : 1.261455 µs
   Execution time mean : 1.261678 µs
   Execution time sample std-deviation : 21.422153 ns
   Execution time std-deviation : 21.849510 ns
   Execution time lower quantile : 1.240191 µs ( 2.5%)
   Execution time upper quantile : 1.293656 µs (97.5%)
   Overhead used : 7.345261 ns
   Found 2 outliers in 60 samples (3.3333 %)
    low-severe   1 (1.6667 %)
    low-mild     1 (1.6667 %)
    Variance from outliers : 6.2807 % Variance is slightly inflated by outliers"
  
====================================================
  
  (c/with-progress-reporting
    (c/bench
      (-> {:foo "bar" :baz [10 20] :fizz {:fuzz "bar"}}
          (assoc :index (.toString 1) :set "set-name"))
      :verbose))
  
  "Evaluation count : 45298980 in 60 samples of 754983 calls.
  Execution time sample mean : 1.345579 µs
  Execution time mean : 1.345796 µs
  Execution time sample std-deviation : 25.551965 ns
  Execution time std-deviation : 26.748618 ns
  Execution time lower quantile : 1.323817 µs ( 2.5%)
  Execution time upper quantile : 1.406625 µs (97.5%)
  Overhead used : 7.345261 ns

  Found 3 outliers in 60 samples (5.0000 %)
  low-severe	 3 (5.0000 %)
  Variance from outliers : 7.8755 % Variance is slightly inflated by outliers"
```

I salute your attention to performance implications, but I think there's no place for an optimization here. A single `assoc` is fine.

Comment (sol-sh) [reply src/main/clojure/aerospike_clj/client.clj:153]: i think the performance diff is big enough a specially this is in a tight loop code
the code i wrote has no disadvantage, as in term in readability its identical as the use of -> is quite idiomatic

**Review by barkanido — APPROVED** [review]

**Comment by barkanido** [comment]
Thanks! will release soon.

**Pull request merged**